src/scene/board-scene-controller.ts

```diff
@@ -130,32 +176,30 @@ export default class BoardSceneController extends SyncedSceneController {
         }
     }
 
-    public onPopUpClick(submit: boolean) {
-        if (!this.currentTurn) return;
+    public onClickSubmitProperty() {
+        this.scene.hideBuyTilePopUp();
+        this.syncProxy.buyProperty(this.aznopoly.uuid);
+    }
 
-        if(this.aznopoly.uuid == this.currentTurn?.getPlayer()) {
-            this.scene.hideBuyTilePopUp();
-            if (submit) {
-                this.currentTurn.doBuyProperty()
-            }
-        }
+    public onClickCancelProperty() {
+        this.scene.hideBuyTilePopUp();
+        this.syncProxy.cancelBuyProperty(this.aznopoly.uuid);
     }
 
-    public onPropertyBuy(uuid: string) {
+    public onCanBuyProperty(uuid: string) {
         const player = this.players.find(p => p.uuid == uuid);
         if (!player) {
-            console.error("Player not found");
```

Comment:
I would keep this log. Since this is an assertion of our logic. If this breaks at any time we should know.